gcloud/monitoring/group.py

```diff
@@ -0,0 +1,507 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Groups for the `Google Stackdriver Monitoring API (V3)`_.
+
+.. _Google Stackdriver Monitoring API (V3):
+    https://cloud.google.com/monitoring/api/ref_v3/rest/v3/\
+    projects.groups
+"""
+
+import re
+
+from gcloud._helpers import _name_from_project_path
+from gcloud.exceptions import NotFound
+from gcloud.monitoring._helpers import _format_timestamp
+from gcloud.monitoring.resource import Resource
+
+
+_GROUP_TEMPLATE = re.compile(r"""
+    projects/            # static prefix
+    (?P<project>[^/]+)   # initial letter, wordchars + hyphen
+    /groups/             # static midfix
+    (?P<name>[^/]+)      # initial letter, wordchars + allowed punc
+""", re.VERBOSE)
+
+
+def group_id_from_name(path, project=None):
+    """Validate a group URI path and get the group ID.
+
+    :type path: string
+    :param path: URI path for a group API request.
+
+    :type project: string or None
+    :param project: The project associated with the request. It is
+                    included for validation purposes.
+
+    :rtype: string
+    :returns: Group ID parsed from ``path``.
+    :raises: :class:`ValueError` if the ``path`` is ill-formed or if
+             the project from the ``path`` does not agree with the
+             ``project`` passed in.
+    """
+    if not path:
+        return ''
```

Comment:
The special case is there to handle the conversion for empty parent_name. But I will remove it for clarity.